### PR TITLE
feat: [PL-30529]: Added org filter in Project selector

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

Changes - 

* Added Orgfilter in project selector
* Created a common component for Org dropdown
* Added search functionality in org dropdown

#### Screenshots
Screenshot 2023-01-12 at 4 05 22 PM

Screenshot 2023-01-16 at 6 50 33 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <DropDown
+      className={props.className}
+      buttonTestId="org-select"
+      onChange={props.onChange}
+      value={props.value}
+      items={orgListPromise}
+      usePortal={true}
+      addClearBtn={true}
+      query={query}
+      onQueryChange={setQuery}
```
Comment (abhinavrastogi-harness) [inline src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx:48]: what about throttling and prior request cancellation?

File: src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const OrgDropdown: React.FC<OrgDropdownProps> = props => {
+  const [query, setQuery] = useState<string>()
+  const { getString } = useStrings()
+
+  function orgListPromise(): Promise<SelectOption[]> {
+    return new Promise<SelectOption[]>(resolve => {
+      getOrganizations({ queryParams: { search_term: query, limit: 100 } })
+        .then(result => {
+          const selectItems = result?.map(item => {
+            return { label: item.org?.name || '', value: item.org?.identifier || '' }
```
Comment (abhinavrastogi-harness) [inline src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx:28]: we  should not show empty name in any case.. if `item.org.name` is empty, skip that item

File: src/modules/45-projects-orgs/pages/projects/ProjectsPage.tsx [hunk 3]
```diff
@@ -170,21 +124,19 @@ const ProjectsListPage: React.FC = () => {
...
+              if (item.value === '') {
                 clearOrgFilterPreference()
-                updateQueryParams({ orgIdentifier: undefined, page: 0 })
               } else {
                 setOrgFilterPreference(item.value as string)
-                updateQueryParams({ orgIdentifier: item.value as string, page: 0 })
               }
+              history.push({
+                pathname: routes.toProjects({ accountId }),
+                search: `?orgIdentifier=${item.value.toString()}`
```
Comment (abhinavrastogi-harness) [inline src/modules/45-projects-orgs/pages/projects/ProjectsPage.tsx:137]: its possible for `item.value` to be empty? in that case we should not add the query param

File: src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/45-projects-orgs/components/OrgDropdown/OrgDropdown.tsx:48]: It's handled in Dropdown component - https://github.com/harness/uicore/blob/60c6c28e2c4a4ea2fc8c921f2846a5b82a29c483/packages/uicore/src/components/DropDown/DropDown.tsx#L174

File: src/modules/45-projects-orgs/pages/projects/ProjectsPage.tsx (hunk 3 above)
Comment (avinashmadhwani02) [reply src/modules/45-projects-orgs/pages/projects/ProjectsPage.tsx:137]: item.value will be empty when User clears the search. 

Ideally we should remove the query param, but we don't have any utility to clear the query param. We are just removing the value of query param in other places too. 


Uploading Screen Recording 2023-01-30 at 5.44.51 PM.mov…

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Pull request merged**